packages/sit-onyx/src/components/TestInput/TestInput.vue

```diff
@@ -120,22 +120,17 @@ const handleChange = (event: Event) => {
 };
 
 watch([value, inputElement], () => {
-  // update validity state when value changes
   if (!inputElement.value) return;
-  validityState.value = inputElement.value.validity;
+  const newValidityState = transformValidityStateToObject(inputElement.value!.validity);
+  // only update + emit the validity state when value changes
+  if (JSON.stringify(newValidityState) !== JSON.stringify(validityState.value)) {
```

Comment:
If I remember JSON.stringify is not be most reliable way to compare objects.
For the TetInput its fine for me to do it like that but we should change it for our real components

Reply:
is this better?

```suggestion
  if (
      !validityState.value ||
    Object.values(newValidityState).toString() != Object.values(validityState.value).toString()
  ) {
```

Reply:
The problem is the same^^ Both rely on that all keys are in the same order so the comparison fails e.g. for `{a: 1, b: 2}` and `{b:2, a:1}` although the objects are the same.

Usually you'd use a package like [fast-deep-equals](https://www.npmjs.com/package/fast-deep-equal) but since we agreed to only have minimal direct dependencies we probably have to implement a minimal version of it on our own that iterates through the objects and compares the content of each key instead of their string representation :)

Reply:
oki, my thought was that the order will stay the same because we iterate on the same pattern through the ordered keys we get from the util. but I was also on the brink of writing a compare util, I'll just do that so we don't have to worry about it in the future :3